ludwig/config_validation/checks.py

```diff
@@ -0,0 +1,327 @@
+        if input_feature[TYPE] in SEQUENCE_OUTPUT_FEATURE_TYPES:
+            has_sequence_input = True
+            break
+    if not has_sequence_input:
+        raise ConfigValidationError(
+            "Sequence concat combiner should only be used for at least one sequential input feature."
+        )
+
+
+@register_config_check
+def check_comparator_combiner_requirements(config: ModelConfigDict) -> None:
+    """Checks that all of the feature names for entity_1 and entity_2 are valid features."""
+    if config[MODEL_TYPE] != MODEL_ECD:
+        return
+    if config[COMBINER] != "comparator":
+        return
+
+    input_feature_names = {input_feature[NAME] for input_feature in config[INPUT_FEATURES]}
+    for entity in ["entity_1", "entity_2"]:
+        for feature_name in config[COMBINER][entity]:
+            if feature_name not in input_feature_names:
+                raise ConfigValidationError(
+                    f"Feature {feature_name} in {entity} for the comparator combiner is not a valid "
+                    "input feature name."
+                )
+
+
+@register_config_check
+def check_class_balance_preprocessing(config: ModelConfigDict) -> None:
+    """Class balancing is only available for datasets with a single output feature."""
+    if config[PREPROCESSING]["oversample_minority"] or config[PREPROCESSING]["undersample_majority"]:
+        if len(config[OUTPUT_FEATURES]) != 1:
+            raise ConfigValidationError("Class balancing is only available for datasets with a single output feature.")
+        if config[OUTPUT_FEATURES][0][TYPE] != BINARY:
+            raise ConfigValidationError("Class balancing is only supported for binary output features.")
+
+
+@register_config_check
+def check_sampling_exclusivity(config: ModelConfigDict) -> None:
+    """Oversample minority and undersample majority are mutually exclusive."""
+    if config[PREPROCESSING]["oversample_minority"] and config[PREPROCESSING]["undersample_majority"]:
+        raise ConfigValidationError(
+            "Oversample minority and undersample majority are mutually exclusive. Specify only one method."
+        )
+
+
+@register_config_check
+def check_validation_metric_exists(config: ModelConfigDict) -> None:
+    """Checks that the specified validation metric exists."""
+    validation_metric_name = config[TRAINER]["validation_metric"]
+
+    # Get all valid metrics.
+    feature_to_metric_names_map = get_feature_to_metric_names_map(config[OUTPUT_FEATURES])
+    all_valid_metrics = set()
+    for metric_names in feature_to_metric_names_map.values():
+        all_valid_metrics.update(metric_names)
+
+    if validation_metric_name not in all_valid_metrics:
+        raise ConfigValidationError(
+            f"User-specified trainer.validation_metric '{validation_metric_name}' is not valid. "
+            f"Available metrics are: {all_valid_metrics}"
+        )
+
+
+@register_config_check
+def check_splitter(config: ModelConfigDict) -> None:
```

Comment:
All these split checks are run after we have generated the `ModelConfig` object, but for some reason we're running them against untyped dicts instead of the config object? We should change these to operate on the `ModelConfig` object for better type safety.